crates/orchestrator/src/tests/config.rs

```diff
@@ -52,128 +99,208 @@ impl Default for TestConfigBuilder {
+        ConfigType::Dummy => Box::new(MockSettlementClient::new()),
+    }
+}
+
+async fn init_prover_client(service: ConfigType) -> Box<dyn ProverClient> {
+    let settings_provider = DefaultSettingsProvider {};
+    match service {
+        ConfigType::Mock(client) => impl_mock_match!(client, MockType::ProverClient, ProverClient),
+        ConfigType::Actual => build_prover_service(&settings_provider),
+        ConfigType::Dummy => Box::new(MockProverClient::new()),
+    }
+}
+
+async fn init_alerts(service: ConfigType) -> Box<dyn Alerts> {
+    match service {
+        ConfigType::Mock(client) => impl_mock_match!(client, MockType::Alerts, Alerts),
+        ConfigType::Actual => build_alert_client().await,
+        ConfigType::Dummy => Box::new(MockAlerts::new()),
+    }
+}
+
+async fn init_storage_client(service: ConfigType) -> Box<dyn DataStorage> {
+    match service {
+        ConfigType::Mock(client) => impl_mock_match!(client, MockType::Storage, Storage),
+        ConfigType::Actual => {
+            let storage = get_storage_client().await;
+            match get_env_var_or_panic("DATA_STORAGE").as_str() {
+                "s3" => storage.as_ref().build_test_bucket(&get_env_var_or_panic("AWS_S3_BUCKET_NAME")).await.unwrap(),
+                _ => panic!("Unsupported Storage Client"),
+            }
+            storage
+        }
+        ConfigType::Dummy => Box::new(MockDataStorage::new()),
+    }
+}
+
+async fn init_queue_client(service: ConfigType) -> Box<dyn QueueProvider> {
+    let aws_config = aws_config::load_from_env().await;
+    match service {
+        ConfigType::Mock(client) => impl_mock_match!(client, MockType::Queue, Queue),
+        ConfigType::Actual => build_queue_client(&aws_config),
+        ConfigType::Dummy => Box::new(MockQueueProvider::new()),
+    }
+}
+
+async fn init_database(service: ConfigType) -> Box<dyn Database> {
+    match service {
+        ConfigType::Mock(client) => impl_mock_match!(client, MockType::Database, Database),
+        ConfigType::Actual => build_database_client().await,
+        ConfigType::Dummy => Box::new(MockDatabase::new()),
+    }
+}
+
+async fn init_starknet_client(service: ConfigType) -> (Arc<JsonRpcClient<HttpTransport>>, Option<MockServer>) {
+    fn get_provider() -> (Arc<JsonRpcClient<HttpTransport>>, Option<MockServer>) {
+        let server = MockServer::start();
+        let port = server.port();
+        let service = Arc::new(JsonRpcClient::new(HttpTransport::new(
+            Url::parse(format!("http://localhost:{}", port).as_str()).expect("Failed to parse URL"),
+        )));
+        (service, Some(server))
+    }
+
+    match service {
+        ConfigType::Mock(client) => {
+            if let MockType::StarknetClient(starknet_client) = client {
+                (starknet_client, None)
+            } else {
+                panic!("Mock client is not a Starknet Client");
+            }
+        }
+        ConfigType::Actual | ConfigType::Dummy => get_provider(),
```

Comment:
for dummy we can just return any random client right? we don't need to really spin up a server

Reply:
Valid, implemented.